base/c.jl

```diff
@@ -96,6 +94,105 @@ convert(::Type{Cstring}, s::Symbol) = Cstring(unsafe_convert(Ptr{Cchar}, s))
 
 # in string.jl: unsafe_convert(::Type{Cwstring}, s::WString)
 
+# FIXME: this should be handled by implicit conversion to Cwstring, but good luck with that
```

Comment:
I forget the details, but `unsafe_convert` and `cconvert` were not cooperative. I spent an hour or so going back and forth with @yuyichao and @vtjnash and it didn't lead to anything workable. That would be the right solution, rather than exposing these internal conversion functions.